hips/hip-0027.md

```diff
@@ -0,0 +1,181 @@
+* Helm does not complain in any way, and
+* a hook resource annotated this way is **never deleted**.
+
+It seems that the current behaviour is a coincidence or mistake, rather than intended.
+
+## Rationale
+
+In line with [Hyrum's Law][Hyrum's Law], at least [one project][StackRox chart] depends on the current behaviour.
+Namely, it specifies a hook deletion policy of `never` on most of its storage-related resources
+(such as `PersistentVolumeClaim`).
+These resources are applied as hooks, in order to guard against data loss,
+see [appendix A](#appendix-a-why-create-storage-resources-as-hooks) if you would like to know more.
+
+I was able to find [one more unrelated chart][Anance personal chart] that also specifies the same value.
+
+Some charts also allow the users to specify the policy using a helm var,
+and it's not clear whether they validate the value before using it in a template.
+
+The current Helm behaviour in this case is unspecified.
+There is a concern that a change in the current implementation
+(for example applying the default policy more aggressively) could cause a catastrophic data loss.
+
+## Specification
+
+This proposal:
+- suggests keeping status quo for helm 3, for the sake of backwards compatibility,
+- outlines a few options for helm 4.
+
+### For Helm 3
+
+- No semantic changes to production code.
+- Add a regression test to make sure that the current behaviour is not changed by mistake.
+  This is probably easiest to do by extracting the defaulting code to a separate unexported function and
+  adding a unit test for the function.
+
+### For Helm 4
+
+There are a few dimensions in which we can make a change, that are somewhat interconnected:
+- whether to accept a deletion disabling policy at all, or break compatibility and reject it,
+- if accepted, whether support it officially (documented, etc), or deprecate it,
+- what to do about (other) undocumented hook deletion policy values,
+
+#### 1. Official support for a policy which disables hook deletion
+
+Add a new, documented hook deletion policy value: `never`.
+When specified, such hook resource is not deleted.
+Effectively, the same as suggestion for Helm 3, but explicitly supported.
+
+Pros:
+- Keeps compatibility with Helm 3.
+- Allows a notion of resources which are installed by helm, but afterwards not managed by it in any way.
+- StackRox chart continues to work as before without changing.
+- Other charts (if any) which happen depend on current undocumented behaviour could easily be made to work by changing
+  whatever value they use to `never`.
+
+Cons:
+- It seems that hooks were generally intended to be resources whose previous instances are cleaned up
+  before (subsequent) chart applications. Supporting this policy breaks this assumption and
+  introduces some complexity when reasoning about possible scenarios. For example such `pre-install`
+  hook resources need to be guarded by a `if $.Release.IsInstall` condition in order not to break upgrades.
```

Comment:
I prefer this one and would call it a bug fix. I think the con you listed, while valid, is probably the expected behavior for anyone that was previously using a not-supported string. This would just make it more explicit.